src/Select/index.js

```diff
@@ -50,6 +52,45 @@ export const getItemByValue = (value, items) => {
   return founditem || "";
 };
 
+/**
+ * @param {Object} item an item from `items`
+ * @param {array} items downshift index `items`
+ * @returns {Number} index of item
+ */
+export const getItemIndex = ({ props }, items) => {
+  const values = items.map(({ props }) => props.value);
+  return values.indexOf(props.value);
+};
+
+/**
+ * @param {Object} highlightedIndex index of currently highlight item
+ * @param {Array} categoryChildren child items in a given category
+ * @param {Array} items downshift `items`
+ * @returns {Boolean} if the provided item is in the category
+ */
+export const isHighlightedInCategory = (
+  highlightedIndex,
+  categoryChildren,
+  items
+) => {
+  if (highlightedIndex < 0) return false;
+  const highlightedValue = items[highlightedIndex].props.value;
+  const categoryValues = categoryChildren.map((child) => child.props.value);
+  return categoryValues.includes(highlightedValue);
+};
+
+/**
+ * @param {Object} selectedItem
+ * @param {Array} categoryChildren child items in a given category
+ * @returns {Boolean} if the selected item is in the given cagetory children
+ */
+export const isSelectedItemInCategory = (selectedItem, categoryChildren) => {
+  if (!selectedItem) return false;
+  const selectedValue = selectedItem.props.value;
+  const categoryValues = categoryChildren.map((child) => child.props.value);
+  return categoryValues.includes(selectedValue);
```

Comment:
This doesn't work in the case that the selected value is a subset of another value
<img width="917" alt="image" src="https://github.com/narmi/design_system/assets/51931432/b2c98e48-63e3-431f-b179-3009778a18a0">